### PR TITLE
undefined behaviour on node delete

### PR DESCRIPTION
**Pull request by kvahed**

### CODE REVIEW

**Comment by kvahed** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/7615/

File: arangod/Agency/Node.cpp [hunk 1]
```diff
@@ -339,6 +339,11 @@ Node& Node::operator()(std::vector<std::string> const& pv) {
   return *current;
 }
 
+std::shared_ptr<Node> Node::child (std::string const& name) {
```
Comment (jsteemann) [inline arangod/Agency/Node.cpp]:
```suggestion
std::shared_ptr<Node> Node::child(std::string const& name) {
```

File: arangod/Agency/Node.cpp [hunk 2]
```diff
@@ -339,6 +331,11 @@ Node& Node::operator()(std::vector<std::string> const& pv) {
...
```
Comment (jsteemann) [inline arangod/Agency/Node.cpp]:
```suggestion
std::shared_ptr<Node> Node::child(std::string const& name) {
```

**Comment by kvahed** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7653/

**Pull request merged**